### ISSUE TITLE
Clarification on 'pretrained=True' behavior when training from a custom .yaml file

### ISSUE THREAD

**Issue by yuchenwu73** [ultralytics/ultralytics#21358]
### Search before asking

- [x] I have searched the Ultralytics YOLO [issues](https://github.com/ultralytics/ultralytics/issues) and [discussions](https://github.com/orgs/ultralytics/discussions) and found no similar questions.

### Search before asking

- [x] I have searched the Ultralytics YOLO [issues](https://github.com/ultralytics/ultralytics/issues) and [discussions](https://github.com/orgs/ultralytics/discussions) and found no similar questions.

### Question

Hello Ultralytics Team and community,

I have a question regarding the precise behavior of the `pretrained` argument when training a model defined by a custom `.yaml` file. This is for a competition with a strict rule that only allows the use of ImageNet pre-trained weights, not other datasets like COCO.

I have created my own model architecture file, `yolo11x-p2.yaml`. My training process starts like this:

```python
from ultralytics import YOLO

# Initialize the model from the yaml architecture file
model = YOLO('yolo11x-p2.yaml') 

# Start training
# model.train(...)
```

My specific questions are:

1.  When I call `model.train(..., pretrained=True)`, is it correct to assume that the framework will automatically download and load the **ImageNet pre-trained weights** for the backbone of my model, while the neck and head are randomly initialized?

2.  Conversely, when I call `model.train(..., pretrained=False)`, does this guarantee that the **entire model** (backbone, neck, and head) is initialized with random weights, with no pre-trained weights being loaded at all?

I would be very grateful for a confirmation of this behavior. Understanding this correctly is critical for me to comply with the competition rules.

Thank you for your fantastic work！

**Comment by UltralyticsAssistant** [comment]
👋 Hello @yuchenwu73, thank you for your detailed question and for your interest in Ultralytics 🚀! For questions about model initialization and training behavior, we recommend visiting the [Docs](https://docs.ultralytics.com/)—especially the sections on [Python](https://docs.ultralytics.com/usage/python/) and [CLI](https://docs.ultralytics.com/usage/cli/) usage, where many common questions are addressed.

This is an automated response to help you as quickly as possible. An Ultralytics engineer will review your question and provide further assistance soon.

If your question relates to a 🐛 Bug Report or unexpected behavior, please provide a [minimum reproducible example](https://docs.ultralytics.com/help/minimum-reproducible-example/) to help us investigate.

For custom training or model architecture ❓ Questions like yours, please share as much detail as possible—such as your YAML file, training logs, and any relevant code snippets. Also, make sure you are following our [Tips for Best Training Results](https://docs.ultralytics.com/guides/model-training-tips/).

Join the Ultralytics community in the way that suits you:
- For real-time chat, head to [Discord](https://discord.com/invite/ultralytics) 🎧
- For in-depth discussions, visit [Discourse](https://community.ultralytics.com/)
- To share and explore community knowledge, check out our [Subreddit](https://www.reddit.com/r/Ultralytics/)

## Upgrade

Please ensure you are using the latest `ultralytics` package and all [requirements](https://github.com/ultralytics/ultralytics/blob/main/pyproject.toml) in a [**Python>=3.8**](https://www.python.org/) environment with [**PyTorch>=1.8**](https://pytorch.org/get-started/locally/):

```bash
pip install -U ultralytics
```

## Environments

YOLO can be run in any of the following up-to-date verified environments (with all dependencies including [CUDA](https://developer.nvidia.com/cuda-zone)/[CUDNN](https://developer.nvidia.com/cudnn), [Python](https://www.python.org/) and [PyTorch](https://pytorch.org/) preinstalled):

- **Notebooks** with free GPU: <a href="https://console.paperspace.com/github/ultralytics/ultralytics"><img src="https://assets.paperspace.io/img/gradient-badge.svg" alt="Run on Gradient"/></a> <a href="https://colab.research.google.com/github/ultralytics/ultralytics/blob/main/examples/tutorial.ipynb"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"></a> <a href="https://www.kaggle.com/models/ultralytics/yolo11"><img src="https://kaggle.com/static/images/open-in-kaggle.svg" alt="Open In Kaggle"></a>
- **Google Cloud** Deep Learning VM. See [GCP Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/google_cloud_quickstart_tutorial/)
- **Amazon** Deep Learning AMI. See [AWS Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/aws_quickstart_tutorial/)
- **Docker Image**. See [Docker Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/docker_image_quickstart_tutorial/) <a href="https://hub.docker.com/r/ultralytics/ultralytics"><img src="https://img.shields.io/docker/pulls/ultralytics/ultralytics?logo=docker" alt="Docker Pulls"></a>

## Status

<a href="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml?query=event%3Aschedule"><img src="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml/badge.svg" alt="Ultralytics CI"></a>

If this badge is green, all [Ultralytics CI](https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml?query=event%3Aschedule) tests are currently passing. CI tests verify correct operation of all YOLO [Modes](https://docs.ultralytics.com/modes/) and [Tasks](https://docs.ultralytics.com/tasks/) on macOS, Windows, and Ubuntu every 24 hours and on every commit.

Thank you again for reaching out! An Ultralytics engineer will respond with more detail on your specific inquiry soon.

**Comment by yuchenwu73** [comment]
Hi, thanks for the automated response.

As requested, here is the custom `yolo11.yaml` file I am using. I'm trying to train a model with a custom architecture.

```yaml
# Ultralytics YOLO 🚀, AGPL-3.0 license
# YOLO11 object detection model with P3-P5 outputs. For Usage examples see https://docs.ultralytics.com/tasks/detect

# Parameters
nc: 80 # number of classes
scales: # model compound scaling constants, i.e. 'model=yolo11n.yaml' will call yolo11.yaml with scale 'n'
  # [depth, width, max_channels]
  n: [0.50, 0.25, 1024] # summary: 319 layers, 2624080 parameters, 2624064 gradients, 6.6 GFLOPs
  s: [0.50, 0.50, 1024] # summary: 319 layers, 9458752 parameters, 9458736 gradients, 21.7 GFLOPs
  m: [0.50, 1.00, 512] # summary: 409 layers, 20114688 parameters, 20114672 gradients, 68.5 GFLOPs
  l: [1.00, 1.00, 512] # summary: 631 layers, 25372160 parameters, 25372144 gradients, 87.6 GFLOPs
  x: [1.00, 1.50, 512] # summary: 631 layers, 56966176 parameters, 56966160 gradients, 196.0 GFLOPs

# YOLO11n backbone
backbone:
  # [from, repeats, module, args]
  - [-1, 1, Conv, [64, 3, 2]] # 0-P1/2
  - [-1, 1, Conv, [128, 3, 2]] # 1-P2/4
  - [-1, 2, C3k2, [256, False, 0.25]]
  - [-1, 1, Conv, [256, 3, 2]] # 3-P3/8
  - [-1, 2, C3k2, [512, False, 0.25]]
  - [-1, 1, Conv, [512, 3, 2]] # 5-P4/16
  - [-1, 2, C3k2, [512, True]]
  - [-1, 1, Conv, [1024, 3, 2]] # 7-P5/32
  - [-1, 2, C3k2, [1024, True]]
  - [-1, 1, SPPF, [1024, 5]] # 9
  - [-1, 2, C2PSA, [1024]] # 10

# YOLO11n head
head:
  - [-1, 1, nn.Upsample, [None, 2, "nearest"]]
  - [[-1, 6], 1, Concat, [1]] # cat backbone P4
  - [-1, 2, C3k2, [512, False]] # 13

  - [-1, 1, nn.Upsample, [None, 2, "nearest"]]
  - [[-1, 4], 1, Concat, [1]] # cat backbone P3
  - [-1, 2, C3k2, [256, False]] # 16 (P3/8-small)

  - [-1, 1, nn.Upsample, [None, 2, "nearest"]]
  - [[-1, 2], 1, Concat, [1]] # cat backbone P2
  - [-1, 2, C3k2, [128, False]] # 19 (P2/4-xsmall)

  - [-1, 1, Conv, [256, 3, 2]]
  - [[-1, 16], 1, Concat, [1]] # cat head P3
  - [-1, 2, C3k2, [256, False]] # 22 (P3/8-medium)

  - [-1, 1, Conv, [256, 3, 2]]
  - [[-1, 13], 1, Concat, [1]] # cat head P4
  - [-1, 2, C3k2, [512, False]] # 25 (P4/16-medium)

  - [-1, 1, Conv, [512, 3, 2]]
  - [[-1, 10], 1, Concat, [1]] # cat head P5
  - [-1, 2, C3k2, [1024, True]] # 28 (P5/32-large)

  - [[19, 22, 25, 28], 1, Detect, [nc]] # Detect(P2, P3, P4, P5)
```

Thank you for your help!

**Comment by Y-T-G** [comment]
> My specific questions are:
>
> 1. When I call `model.train (..., pretrained = True)`, is it correct to assume that the framework will automatically download and load the **ImageNet pre-trained weights** for the backbone of my model, while the neck and head are randomly initialized?

`pretrained=True` on a model initialized from `.yaml,` doesn't do anything. It doesn't load any pretrained weights. Setting it `True` or `False`, the effect would be the same.

> 2. Conversely, when I call `model.train (..., pretrained = False)`, does this guarantee that the **entire model** (backbone, neck, and head) is initialized with random weights, with no pre-trained weights being loaded at all?

A model loaded from `.yaml` is always randomly initialized regardless of `pretrained` being `True` or `False`. Only setting `pretrained` to a path to another checkpoint makes a difference which would cause the weights from that checkpoint to be loaded.
